tests/suites/test_suite_ssl.function

```diff
@@ -104,6 +111,20 @@ void init_handshake_options( handshake_test_options *opts )
   opts->srv_log_fun = NULL;
   opts->cli_log_fun = NULL;
   opts->resize_buffers = 1;
+#if defined(MBEDTLS_SSL_CACHE_C)
+  opts->cache = malloc( sizeof( mbedtls_ssl_cache_context ) );
```

Comment:
Please do not call `malloc`. Even test code must use `mbedtls_calloc` and `mbedtls_free` which may be something other than the platform's allocator. (I filed https://github.com/Mbed-TLS/mbedtls/issues/6000 to enforce this on the CI.)

In test code, favor the macro `ASSERT_ALLOC` over direct use of `mbedtls_calloc` where applicable.